src/com/google/enterprise/adaptor/filenet/FnDocumentProperties.java

```diff
@@ -78,293 +76,165 @@ public FnDocumentProperties(Document doc) {
     Property prop = metas.get(name);
     if (prop == null) {
       logger.log(Level.FINEST, "Property not found: {0}", name);
       return;
     }
-    if (prop instanceof PropertyString
-        || prop instanceof PropertyStringList) {
-      logger.log(Level.FINEST, "Getting String property: [{0}]", name);
-      getPropertyStringValue(name, list);
-    } else if (prop instanceof PropertyBinary
-        || prop instanceof PropertyBinaryList) {
-      logger.log(Level.FINEST, "Binary property [{0}] is not supported.", name);
-    } else if (prop instanceof PropertyBoolean
-        || prop instanceof PropertyBooleanList) {
-      logger.log(Level.FINEST, "Getting Boolean property: [{0}]", name);
-      getPropertyBooleanValue(name, list);
-    } else if (prop instanceof PropertyDateTime
-        || prop instanceof PropertyDateTimeList) {
-      logger.log(Level.FINEST, "Getting Date property: [{0}]", name);
-      getPropertyDateValue(name, list);
-    } else if (prop instanceof PropertyFloat64
-        || prop instanceof PropertyFloat64List) {
-      logger.log(Level.FINEST, "Getting Double/Float property: [{0}]", name);
-      getPropertyDoubleValue(name, list);
-    } else if (prop instanceof PropertyInteger32
-        || prop instanceof PropertyInteger32List) {
-      logger.log(Level.FINEST, "Getting Integer/Long property: [{0}]", name);
-      getPropertyLongValue(name, list);
-    } else if (prop instanceof PropertyId
-        || prop instanceof PropertyIdList) {
-      logger.log(Level.FINEST, "Getting Id property: [{0}]", name);
-      getPropertyGuidValue(name, list);
+    if (prop instanceof PropertyString) {
+      getStringValue(name, prop, list);
+    } else if (prop instanceof PropertyStringList) {
+      getStringListValue(name, prop, list);
+    } else if (prop instanceof PropertyBoolean) {
+      getBooleanValue(name, prop, list);
+    } else if (prop instanceof PropertyBooleanList) {
+      getBooleanListValue(name, prop, list);
+    } else if (prop instanceof PropertyDateTime) {
+      getDateValue(name, prop, list);
+    } else if (prop instanceof PropertyDateTimeList) {
+      getDateListValue(name, prop, list);
+    } else if (prop instanceof PropertyFloat64) {
+      getDoubleValue(name, prop, list);
+    } else if (prop instanceof PropertyFloat64List) {
+      getDoubleListValue(name, prop, list);
+    } else if (prop instanceof PropertyInteger32) {
+      getLongValue(name, prop, list);
+    } else if (prop instanceof PropertyInteger32List) {
+      getLongListValue(name, prop, list);
+    } else if (prop instanceof PropertyId) {
+      getGuidValue(name, prop, list);
+    } else if (prop instanceof PropertyIdList) {
+      getGuidListValue(name, prop, list);
     } else {
-      logger.log(Level.FINEST, "Property type for {0} is not determined: ",
+      logger.log(Level.FINEST, "Property type is not supported: {0}",
           prop.getClass().getName());
```

Comment:
I like this very much vs. the RuntimeException that was thrown.
Much better to say "Not this thing. no ... no" than to bomb completely out.